src/storage.js

```diff
@@ -0,0 +1,44 @@
+// Mimic localStorage API with chrome.storage.
+// See also: https://developer.chrome.com/docs/extensions/reference/storage/
+const storage = {
+    getAllItems: () => new Promise(resolve => {
+        chrome.storage.local.get(null, (result) => {
+            resolve(result);
+        });
+    }),
+    // Gets one or more items from storage.
+    getItem: key => new Promise(resolve => {
+        chrome.storage.local.get(key, (result) => {
+            if (result) {
+                resolve(result[key]);
+            } else {
+                resolve(null);
+            }
+        });
+    }),
+    // Set signle key-value pair item.
+    setItem: (key, val) => new Promise(resolve => {
+        chrome.storage.local.set({
+            [key]: val
+        }, resolve)
+    }),
+    // Removes one or more items from storage.
+    removeItem: key => new Promise(resolve => {
+        chrome.storage.local.remove(key);
+        resolve()
+    }),
+};
+
+// A helper function to migrate localStorage to chrome.storage API.
+async function migrateLocalStorage(key) {
```

Comment:
This function is not used within core. Is this intended to be used by the downstream extensions if they need it?

Reply:
Yes. I think it may helpful for downstream extensions. So prefer to keep this. 
For example, my Rust extension needs this function:
https://github.com/huhu/rust-search-extension/blob/master/extension/migration.js#L22-L25